lib/driver.js

```diff
@@ -83,20 +74,54 @@ class AndroidDriver extends BaseDriver {
   }
 
   async initAndroidBootstrap () {
-    this.bootstrap = new Bootstrap(DEVICE_PORT);
+    this.bootstrap = new Bootstrap(DEVICE_PORT, this.opts.websocket);
 
-    await this.bootstrap.start(this.opts.appPackage, false);
+    await this.bootstrap.start(this.opts.appPackage, this.opts.disableAndroidWatchers);
   }
 
   async startAUT () {
     await this.adb.startApp({
       pkg: this.opts.appPackage,
       activity: this.opts.appActivity,
-      action: "android.intent.action.MAIN",
-      category: "android.intent.category.LAUNCHER",
-      flags: "0x10200000"
+      action: this.opts.intentAction,
+      category: this.opts.intentCategory,
+      flags: this.opts.intentFlags,
+      waitPkg: this.opts.appWaitPackage,
+      waitActivity: this.opts.appWaitActivity,
+      optionalIntentArguments: this.opts.optionalIntentArguments,
+      stopApp: this.opts.stopAppOnReset || !this.opts.dontStopAppOnReset
+      pkg: this.opts.appPackage,
+      activity: this.opts.appActivity,
     });
   }
+
+  async checkAppPresent () {
+    logger.debug("Checking whether app is actually present");
+    if (!(await fs.exists(this.opts.app))) {
+      logger.errorAndThrow(`Could not find app apk at ${this.opts.app}`);
+    }
+  }
+
+  async pushStrings (language) {
```

Comment:
Can you add a comment for this function? I'm not really sure how it works, or why it's important.